bigflow/cli.py

```diff
@@ -127,7 +129,9 @@ def execute_workflow(root_package: Path, workflow_id: str, runtime=None):
 
     @param runtime: str determine partition that will be used for write operations.
     """
-    find_workflow(root_package, workflow_id).run(runtime)
+    w = find_workflow(root_package, workflow_id)
```

Comment:
I don't like the idea that workflow initializes logging. It already has 3 responsibilities:

* running jobs
* building execution order
* holding DAG configuration

I think you can move the logging initialization to a function, and the Workflow class can act only as logging_configuration (and probably other kinds of configuration) holder.

What do you think?

Reply:
I don't like initialization in the Workflow. Also user can install bigflow without our `log` module so this code will throw Exception

Reply:
On the other hand, if you want to run a workflow without the CLI, and still have the logging initialized, it's problematic...

Reply:
In fact it is already in a separate function, Workflow just expose wrapper for it.
Agree, that it is better not to add such logic to Workflow.
Changed: Dropped `Workflow.init_logging()`, expose attribute `Workflow.log_config`.